android/src/main/java/com/gr4vy/embedreactnative/EmbedReactNativeModule.java

```diff
@@ -120,6 +144,36 @@ private static WritableArray convertCartItemsToJson(ReadableArray cartItemsArray
       cartItemWritableMap.putString("name", cartItemMap.getString("name"));
       cartItemWritableMap.putInt("quantity", cartItemMap.getInt("quantity"));
       cartItemWritableMap.putInt("unitAmount", cartItemMap.getInt("unitAmount"));
+
+      List<String> optionalProps = new ArrayList<String>(Arrays.asList(
```

Comment:
We only want to pass these values if defined, otherwise `null`

Reply:
You could use a HashMap instead

Reply:
Actually, if we only want that list to iterate over it I think it should be enough with the `Arrays.asList`, or even just a static array:
```java
String[] optionalProps = new String[] {  
  "discountAmount",
  "taxAmount",
  "externalIdentifier",
  "sku",
  "productUrl",
  "imageUrl",
  "categories",
  "productType"
};  
```

Reply:
noice, thanks that's better